src/ThreeEditor/js/commands/DuplicateObjectCommand.ts

```diff
@@ -0,0 +1,106 @@
+import { ObjectManagementFactory } from '../../commands/factories/ObjectManagementFactory';
+import { SimulationSceneChild } from '../../Simulation/Base/SimulationContainer';
+import { SimulationElement } from '../../Simulation/Base/SimulationElement';
+import { isSimulationZone } from '../../Simulation/Base/SimulationZone';
+import { isDetector } from '../../Simulation/Detectors/Detector';
+import { isBasicFigure } from '../../Simulation/Figures/BasicFigures';
+import { isScoringFilter } from '../../Simulation/Scoring/ScoringFilter';
+import { isOutput } from '../../Simulation/Scoring/ScoringOutput';
+import { isQuantity } from '../../Simulation/Scoring/ScoringQuantity';
+import { Command, CommandJSON } from '../Command';
+import { YaptideEditor } from '../YaptideEditor';
+
+interface DuplicateObjectCommandJSON extends CommandJSON {
+	objectUuid: string;
+	actionCommands: CommandJSON[];
+}
+
+export const canBeDuplicated = (object: unknown) => {
+	return (
+		isBasicFigure(object) ||
+		isOutput(object) ||
+		isQuantity(object) ||
+		isScoringFilter(object) ||
+		isDetector(object) ||
+		isSimulationZone(object)
+	);
+};
```

Comment:
This type of method in our project usually uses boolean flags implemented for all simulation elements.
For example https://github.com/yaptide/ui/blob/42647ef4cb8e1b41905e531df568ebd58e6e622e/src/util/hooks/useKeyboardEditorControls.ts#L19-L28

Shouldn't we continue this technique? Or at least make it easier when creating a new Simulation Element to manage whether it can or cannot be cloned. Because right now you have to search for the implementation of the cloning feature to add another condition to the guard function.